httpipfs.go

```diff
@@ -119,6 +130,23 @@ func (hi *HttpIpfs) ServeHTTP(res http.ResponseWriter, req *http.Request) {
 
 	selNode := unixfsnode.UnixFSPathSelectorBuilder(path.String(), dagScope.TerminalSelectorSpec(), false)
 
+	sig := RequestSignature{
+		requestId: req.Header.Get("X-Signature"),
+		cid:       rootCid.String(),
+		protocol:  "https",
+	}
+	b, err := json.Marshal(sig)
+	if err != nil {
+		logError(http.StatusInternalServerError, err)
+		return
+	}
+	sigSigned, err := hi.privKey.Sign(b)
+	if err != nil {
+		logError(http.StatusInternalServerError, err)
+		return
+	}
+	res.Header().Set("X-Signature", string(sigSigned))
```

Comment:
Let's use `X-Attestation` or `X-Retrieval-Attestation` since the header value contains more data than the signature.

```suggestion
	res.Header().Set("X-Attestation", string(sigSigned))
```